### ISSUE TITLE
[youtube] Download speeds still throttled for some clients after range fix

### ISSUE THREAD

**Issue by superislam** [yt-dlp/yt-dlp#6400]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.02.17** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

download is slow when downloading as audio and fast as video

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-x', '--audio-quality', '0', '--add-metadata', 'https://www.youtube.com/watch?v=pJ0auP7dbcY']
[debug] Encodings: locale cp1256, fs utf-8, pref cp1256, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.02.17 [a0a7c01] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2022-11-03-git-5ccd4d3060-full_build-www.gyan.dev (setts), ffprobe 2022-11-03-git-5ccd4d3060-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.02.17, Current version: 2023.02.17
yt-dlp is up to date (2023.02.17)
[youtube] Extracting URL: https://www.youtube.com/watch?v=pJ0auP7dbcY
[youtube] pJ0auP7dbcY: Downloading webpage
[youtube] pJ0auP7dbcY: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] pJ0auP7dbcY: Downloading 1 format(s): 251
[debug] Invoking http downloader on "https://rr2---sn-uxaxjvhxbt2u-xhtz.googlevideo.com/videoplayback?expire=1677815867&ei=2xsBZMmQEYqrxwK3x7SQDQ&ip=156.209.131.29&id=o-AP6fDpGS2eg_Bpej8argoWMtHOiqLzZKBxGxsndnCgwr&itag=251&source=youtube&requiressl=yes&mh=fD&mm=31%2C29&mn=sn-uxaxjvhxbt2u-xhtz%2Csn-aigl6ney&ms=au%2Crdu&mv=m&mvi=2&pl=19&initcwndbps=191250&spc=H3gIhkkFhilEy37sYHuXiZnT_lMxF-k&vprv=1&svpuc=1&mime=audio%2Fwebm&gir=yes&clen=143903469&dur=10988.181&lmt=1674553794508873&mt=1677793993&fvip=1&keepalive=yes&fexp=24007246&c=ANDROID&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRAIgM2Qu7Ls0Opq_HeGnsFCLfyVX4XY5BSuDoRiRAky6GrsCICR96HrEqa5CL66bTy0obVBQUfXX2YeGL2pJfx6fbY67&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhALwiU8fuCUDbk0OBCW9E-BFf_keBgT-yAzRRKat9DlTgAiAGLjZQ0zz8xIcZZ6Sc69t7DdMgkxY08pAU7LxFr8q3Ew%3D%3D"
[download] Resuming download at byte 2195705
[debug] File locking is not supported. Proceeding without locking
[download] Destination: كيف تنجح العلاقات مع ياسر الحزيمي ｜ بودكاست فنجان [pJ0auP7dbcY].webm
[download]   1.9% of  137.24MiB at   25.28KiB/s ETA 01:30:51
ERROR: Interrupted by user
't' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by vphantom** [comment]
Interesting, I'm stuck roughly around 25 KiB/s too today, especially for format 251. 🤔

**Comment by dude22072** [comment]
Also been having issues with all youtube audio formats the past two days. Running a modified version of TheFrenchGhosty's archive script; comments, live chat, video formats, etc all download at a decent speed but audio is always under 30KiB/s despite speedtest showing me as above gigabit download speed.

**Comment by gamer191** [comment]
duplicate of #6369
It will be fixed in the next release
According to https://github.com/yt-dlp/yt-dlp/issues/6369#issuecomment-1450068901
> Until then, you have many options, including using [Daily builds](https://github.com/ytdl-patched/yt-dlp/releases/latest) or [PiP installing from master](https://github.com/yt-dlp/yt-dlp/wiki/Installation#:~:text=master%20branch)

**Comment by Vangelis66** [comment]
If I may just step in 😉 ...

While it might initially appear to the devs as indeed a **duplicate** of #6369, it **actually isn't**; most evil Google are monitoring this repo and have now come with their "own fix", to nullify `yt-dlp`'s committed fixes... 

The latest `yt-dlp` daily build is [2023.03.01.19419](https://github.com/ytdl-patched/yt-dlp/releases/tag/2023.03.01.19419); this was compiled from `yt-dlp` snapshot https://github.com/yt-dlp/yt-dlp/commit/5b28cef72db3b531680d89c121631c73ae05354f, which includes the "fixes" for #6369, f34804b2f920f62a6e893a14a9e2a2144b14dd23 + 5038f6d713303e0967d002216e7a88652401c22a

Yet: 
```shell
yt-dlp -vf 140 "p7FCgw_GlWc" => 

[debug] Command-line config: ['--ffmpeg-location', '..\\FFmpeg', '--downloader-args', 'ffmpeg:-v 8 -stats', '-vf', '140', 'p7FCgw_GlWc']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] ytdl-patched/yt-dlp version 2023.03.01.19419 [9b7cf00] (win_x86_exe)
[debug] ** This build is unofficial daily builds, provided for ease of use.
[debug] ** Please do not ask for any support.
[debug] Python 3.7.9 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL 1.1.1g  21 Apr 2020)
[debug] exe versions: ffmpeg n5.2-dev-2245-N-109649-gab8cde6 (setts), ffprobe n5.2-dev-2245-N-109649-gab8cde6, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: AGB+NSIG (YoutubeIE)
[debug] Plugin directories: ['<redacted>\\yt-dlp\\yt-dlp-plugins\\YTAgeGateBypass.zip\\yt_dlp_plugins', '<redacted>\\yt-dlp\\yt-dlp-plugins\\YTNSigProxy.zip\\yt_dlp_plugins']
[debug] Loaded 1782 extractors
[youtube+AGB+NSIG] Extracting URL: p7FCgw_GlWc
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading webpage
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading android player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading tv embedded player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded client config
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading player 7862ca1f
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded player API JSON
[debug] [youtube+AGB+NSIG] Extracting signature function js_7862ca1f_105
[debug] Loading youtube-sigfuncs.js_7862ca1f_105 from cache
[debug] Loading youtube-nsig.7862ca1f from cache
[debug] [youtube+AGB+NSIG] Decrypted nsig oPGW9c82dzKHz-U => 0-VAXYzMO_3yT
[debug] Loading youtube-nsig.7862ca1f from cache
[debug] [youtube+AGB+NSIG] Decrypted nsig wR-57cpazAlyJ0k => JqOKDJTXM_pG8
[debug] [youtube+AGB+NSIG] Extracting signature function js_7862ca1f_101
[debug] Loading youtube-sigfuncs.js_7862ca1f_101 from cache
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] p7FCgw_GlWc: Downloading 1 format(s): 140
[debug] Invoking dashsegments downloader on "https://rr2---sn-4vguioxu-n3bl.googlevideo.com/videoplayback?expire=1677820864&ei=YC8BZPTXEISp1wLUwoK4Cg&ip=46.12.96.248&id=o-AN-mV5Wv9YUp9ezgeFxbuqle6Q96XRuvc6TGWrYpJyF6&itag=140&source=youtube&requiressl=yes&mh=a3&mm=31%2C29&mn=sn-4vguioxu-n3bl%2Csn-nv47lnsk&ms=au%2Crdu&mv=m&mvi=2&pl=18&gcr=gr&initcwndbps=693750&vprv=1&mime=audio%2Fmp4&ns=f61flAuXtRGkA8HpLjYOoDQL&gir=yes&clen=10309815&dur=636.992&lmt=1637451611628671&mt=1677798791&fvip=4&keepalive=yes&fexp=24007246&c=TVHTML5_SIMPLY_EMBEDDED_PLAYER&txp=5532434&n=JqOKDJTXM_pG8&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgQKzJR9gDv3-gj4E7c-ShxYqHSKT4rWQYptoFmAnWIysCIQDYsFq_cgVkIJbYw_BOpr66HJoPo1B5SAMQ5VO8x04fgw%3D%3D&sig=AOq0QJ8wRAIgMwdZJilDcFCVw7v1tBifs_obb7seo0K1h40l5QrEbLECIDrfO9Y5ESKlRjZXLeEIJ2OkNtoZ-UJ5uE7exVvgnYEw"
[dashsegments] Total fragments: 1
[download] Destination: Kanye West - Famous [p7FCgw_GlWc].m4a
[debug] File locking is not supported. Proceeding without locking
[download]   2.2% of ~   9.83MiB at   50.45KiB/s ETA 03:15 (frag 0/1)
ERROR: Interrupted by user
Terminate batch job (Y/N)?
```
@gamer191, please reopen this issue - and **pukkanda**n must hold off the imminent stable release, for it'll be BROKEN too, if this latest gimmick by Google 😡 isn't mitigated!

Kind regards, thanks in advance!

**Comment by bashonly** [comment]
I can reproduce the audio format throttling on master as @Vangelis66 describes

**Comment by Vangelis66** [comment]
> I can reproduce the **audio format throttling** on `master`

Most sadly, **it isn't limited to just audio** (though `audio-only` was the first I tried...); Google have now completely "broken" `yt-dlp` for `video-only`, too: 
```shell
yt-dlp -vf 399 "p7FCgw_GlWc" => 

[debug] Command-line config: ['--ffmpeg-location', '..\\FFmpeg', '--downloader-args', 'ffmpeg:-v 8 -stats', '-vf', '399', 'p7FCgw_GlWc']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] ytdl-patched/yt-dlp version 2023.03.01.19419 [9b7cf00] (win_x86_exe)
[debug] ** This build is unofficial daily builds, provided for ease of use.
[debug] ** Please do not ask for any support.
[debug] Python 3.7.9 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL 1.1.1g  21 Apr 2020)
[debug] exe versions: ffmpeg n5.2-dev-2245-N-109649-gab8cde6 (setts), ffprobe n5.2-dev-2245-N-109649-gab8cde6, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: AGB+NSIG (YoutubeIE)
[debug] Plugin directories: ['D:\\Vangelis\\Installers\\youtube-dl\\yt-dlp\\yt-dlp-plugins\\YTAgeGateBypass.zip\\yt_dlp_plugins', 'D:\\Vangelis\\Installers\\youtube-dl\\yt-dlp\\yt-dlp-plugins\\YTNSigProxy.zip\\yt_dlp_plugins']
[debug] Loaded 1782 extractors
[youtube+AGB+NSIG] Extracting URL: p7FCgw_GlWc
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading webpage
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading android player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading tv embedded player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded client config
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading player 7862ca1f
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded player API JSON
[debug] [youtube+AGB+NSIG] Extracting signature function js_7862ca1f_101
[debug] Loading youtube-sigfuncs.js_7862ca1f_101 from cache
[debug] Loading youtube-nsig.7862ca1f from cache
[debug] [youtube+AGB+NSIG] Decrypted nsig vjEprBVjAvtOGh8 => -eKiTHxf4oqPE
[debug] [youtube+AGB+NSIG] Extracting signature function js_7862ca1f_105
[debug] Loading youtube-sigfuncs.js_7862ca1f_105 from cache
[debug] Loading youtube-nsig.7862ca1f from cache
[debug] [youtube+AGB+NSIG] Decrypted nsig JxGjhc8vVETpT27 => 5KvUjGz9W_L_T
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] p7FCgw_GlWc: Downloading 1 format(s): 399[debug] Invoking dashsegments downloader on "https://rr2---sn-4vguioxu-n3bl.googlevideo.com/videoplayback?expire=1677823917&ei=TTsBZPyjFeKF6dsPvOiPmAo&ip=46.12.96.248&id=o-ALuYwiFyIpw-zI9gtOOVSDfLzFk4mBJGkHLoiuGvuTPf&itag=399&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&mh=a3&mm=31%2C29&mn=sn-4vguioxu-n3bl%2Csn-nv47zn7y&ms=au%2Crdu&mv=m&mvi=2&pl=18&gcr=gr&initcwndbps=785000&vprv=1&mime=video%2Fmp4&ns=09Ur6WP4o29mc6iXerb6CIAL&gir=yes&clen=140099890&dur=636.902&lmt=1622078607328044&mt=1677801912&fvip=1&keepalive=yes&fexp=24007246&c=TVHTML5_SIMPLY_EMBEDDED_PLAYER&txp=5436432&n=5KvUjGz9W_L_T&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgMc14--0-gj34yNGqB6ZJxhIqUMdJpceqkWc4_xQzSAoCIBodkacLAcGaeaAjzIF4oZVj9Ht0x2Cl73rbir9pPW5g&sig=AOq0QJ8wRgIhAJLG_QQ8TjKGXzs3etmnL7L1nDCqm6qN3ft7EeyR-yWwAiEA8mHUC-Fm4h0N3_YZSxSw2mRgkxAvwoTjiz236KGqFHc%3D"
[dashsegments] Total fragments: 14
[download] Destination: Kanye West - Famous [p7FCgw_GlWc].mp4
[debug] File locking is not supported. Proceeding without locking
[download]   0.1% of ~ 140.00MiB at   37.45KiB/s ETA 01:03:45 (frag 0/14)
ERROR: Interrupted by user
Terminate batch job (Y/N)?
```
It's rather obvious that **Google** have unleashed a **full-fledged war against this (and similar) communities**... Guess the issue's title has to be changed once again 😠 ...

**Comment by Vangelis66** [comment]
> Download speeds still throttled **for some formats** after range fix

Well, I'd make it myself "**for ALL formats**", since now **even the non-segmented ones** (**18** and/or **22**) are being throttled (at least here, YMMV?): 
```shell
yt-dlp -f 18 "p7FCgw_GlWc" => 

[youtube+AGB+NSIG] Extracting URL: p7FCgw_GlWc
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading webpage
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading android player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading tv embedded player API JSON
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded client config
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading player 7862ca1f
[youtube+AGB+NSIG] p7FCgw_GlWc: Downloading web embedded player API JSON
[info] p7FCgw_GlWc: Downloading 1 format(s): 18
[download] Destination: Kanye West - Famous [p7FCgw_GlWc].mp4
[download]   4.4% of   12.33MiB at   83.07KiB/s ETA 02:25
ERROR: Interrupted by user
```
I really wonder what they came up with this time 😠 ... Perhaps new player `7862ca1f` is somehow relevant?

**Comment by bashonly** [comment]
I still get unthrottled speeds with some videos/formats, e.g. the OP's example URL
```
$ yt-dlp -f bv pJ0auP7dbcY
[youtube] Extracting URL: pJ0auP7dbcY
[youtube] pJ0auP7dbcY: Downloading webpage
[youtube] pJ0auP7dbcY: Downloading android player API JSON
[info] pJ0auP7dbcY: Downloading 1 format(s): 313
[dashsegments] Total fragments: 642
[download] Destination: 20221116_UCwjLh640nGXSGa9iHRS31ag_pJ0auP7dbcY__.webm
[download]   1.1% of ~   6.27GiB at   13.15MiB/s ETA 07:42 (frag 7/642)^C
ERROR: Interrupted by user
```

**Comment by gamer191** [comment]
1. OP isn't on master. Theoretically Vangelis66 should have created a new issue, but it's probably easier to just stick to this issue now
2. Vangelis66's url is an age-restricted music video. I'm not at home right now, but it wouldn't surprise me if only music videos, or only age-restricted videos, are throttled on master
3. @bashonly are you sure that the format is relevant, as opposed to my assumption that some videos are fully throttled and others aren't throttled at all?

**Comment by Vangelis66** [comment]
Non age-gated **music** clip, non-segmented format: 
```
yt-dlp -f 22 "o8Y_uR6ZDxo" => 
[youtube+AGB+NSIG] Extracting URL: o8Y_uR6ZDxo
[youtube+AGB+NSIG] o8Y_uR6ZDxo: Downloading webpage
[youtube+AGB+NSIG] o8Y_uR6ZDxo: Downloading android player API JSON
[info] o8Y_uR6ZDxo: Downloading 1 format(s): 22
[download] Destination: ROY KNOX x WTCHOUT - Shadows (Feat. Svniivan) [NCS Release] [o8Y_uR6ZDxo].mp4
[download]  43.9% of   31.68MiB at    1.19MiB/s ETA 00:14
ERROR: Interrupted by user
```
i.e. not throttled!

Same clip, `audio-only` format: 
```shell
yt-dlp -f ba "o8Y_uR6ZDxo" => 

[youtube+AGB+NSIG] Extracting URL: o8Y_uR6ZDxo
[youtube+AGB+NSIG] o8Y_uR6ZDxo: Downloading webpage
[youtube+AGB+NSIG] o8Y_uR6ZDxo: Downloading android player API JSON
[info] o8Y_uR6ZDxo: Downloading 1 format(s): 251
[dashsegments] Total fragments: 1
[download] Destination: ROY KNOX x WTCHOUT - Shadows (Feat. Svniivan) [NCS Release] [o8Y_uR6ZDxo].webm
[download]  76.9% of ~   4.03MiB at    1.07MiB/s ETA 00:00 (frag 0/1)
ERROR: Interrupted by user
```
i.e. not throttled!

So, **gamer191** is on to something 👍 ...

**Comment by gamer191** [comment]
NoCopyrightSounds don't count as music videos. Try with a Vevo video

**Comment by bashonly** [comment]
> are you sure that the format is relevant, as opposed to my assumption that some videos are fully throttled and others aren't throttled at all?

It does seem to be specifically age-gated videos.

No throttling for this VEVO, non-age-gated music video: https://www.youtube.com/watch?v=bRChz-OYi9o

While all of the actually age-gated music videos in this playlist are throttled:
https://www.youtube.com/watch?v=&list=PLl5Tkoh1UlTgdGlIbtcdn8S9vPMEdXaXB

**Comment by Vangelis66** [comment]
> Try with a **Vev**o video

```shell
yt-dlp -f ba "XXYlFuWEuKI" => 

[youtube+AGB+NSIG] Extracting URL: XXYlFuWEuKI
[youtube+AGB+NSIG] XXYlFuWEuKI: Downloading webpage
[youtube+AGB+NSIG] XXYlFuWEuKI: Downloading android player API JSON
[info] XXYlFuWEuKI: Downloading 1 format(s): 251
[dashsegments] Total fragments: 1
[download] Destination: The Weeknd - Save Your Tears (Official Music Video) [XXYlFuWEuKI].webm
[download]  50.3% of ~   3.97MiB at    1.24MiB/s ETA 00:01 (frag 0/1)
ERROR: Interrupted by user
```
=> NOT throttled... 

So it appears only **the age-gated ones** are being throttled in `master`... Yet, it's a **recent change by Google**...

**Comment by gamer191** [comment]
@superislam does it work with https://github.com/ytdl-patched/yt-dlp/releases? If not, please post a verbose log
If yes, then it's issue #6369, which will be fixed in the next release

**Comment by Vangelis66** [comment]
@superislam : As **gamer191** pointed out, you're not on `master` or **latest daily build**; I am, thus: 
```shell
yt-dlp -f ba "pJ0auP7dbcY" => 

[youtube+AGB+NSIG] Extracting URL: pJ0auP7dbcY
[youtube+AGB+NSIG] pJ0auP7dbcY: Downloading webpage
[youtube+AGB+NSIG] pJ0auP7dbcY: Downloading android player API JSON
[info] pJ0auP7dbcY: Downloading 1 format(s): 251
[dashsegments] Total fragments: 14
[download] Destination: كيف تنجح العلاقات مع ياسر الحزيمي ｜ بودكاست فنجان [pJ0auP7dbcY].webm
[download]   2.3% of ~ 140.00MiB at    1.14MiB/s ETA 01:59 (frag 0/14)
ERROR: Interrupted by user
```

**Comment by superislam** [comment]
> @superislam does it work with https://github.com/ytdl-patched/yt-dlp/releases? If not, please post a verbose log If yes, then it's issue #6369, which will be fixed in the next release

yes resolved

**Comment by gamer191** [comment]
1. Are age-gated non-music videos throttled?
2. I'd be interested what happens if you pass cookies, including for accounts that don't have access to age-gated videos. If you test this, please also include whether or not you have youtube premium

**Comment by fren-archivist** [comment]
I am also seeing this issue on a number of videos which are not age-restricted for both audio and video formats. I am using the patched version 2023.03.01.19419.  It is a problem on most new videos for me, though somewhat inconsistent: sometimes only one format is throttled, sometimes both are, sometimes restarting the download solves the problem but sometimes it does not. Here is one example, I could give many more if it would help:

```
[debug] Command-line config: ['sJrDp5UeRIM', '-f', '243', '-vU']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] ytdl-patched/yt-dlp version 2023.03.01.19419 [9b7cf00] (win_exe)
[debug] ** This build is unofficial daily builds, provided for ease of use.
[debug] ** Please do not ask for any support.
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg git-2020-07-29-cbb6ba2, ffprobe git-2020-07-29-cbb6ba2, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Fetching release info: https://api.github.com/repos/ytdl-patched/yt-dlp/releases/latest
Latest version: 2023.03.01.19419, Current version: 2023.03.01.19419
yt-dlp is up to date (2023.03.01.19419)
[youtube] Extracting URL: sJrDp5UeRIM
[youtube] sJrDp5UeRIM: Downloading webpage
[youtube] sJrDp5UeRIM: Downloading android player API JSON
[debug] Loading youtube-nsig.7862ca1f from cache
[debug] [youtube] Decrypted nsig XDya9AyvDaxN9kq => Lx_OTmr0Ont_h
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] sJrDp5UeRIM: Downloading 1 format(s): 243
[debug] Invoking dashsegments downloader on "https://rr2---sn-p5qlsn76.googlevideo.com/videoplayback?expire=1677833549&ei=7WABZLG3K83s8wSszp3ACQ&ip=xxxxxxxxxxxxxxxx&id=o-ALQR5Tc9LclKNRkCP0U14EMmqSq_bUZPUxJ-yvitqi16&itag=243&aitags=134%2C136%2C160%2C243%2C298%2C299&source=youtube&requiressl=yes&mh=pX&mm=31%2C29&mn=sn-p5qlsn76%2Csn-p5qddn76&ms=au%2Crdu&mv=m&mvi=2&pl=37&initcwndbps=1751250&vprv=1&mime=video%2Fwebm&ns=IDegC3IIl28mjUOvUa9WeV8L&gir=yes&clen=334865871&dur=8983.000&lmt=1677770505967956&mt=1677811254&fvip=4&keepalive=yes&fexp=24007246&c=WEB&txp=7219224&n=Lx_OTmr0Ont_h&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRAIgbRIkjqkBn2S9pNFaCgs2uXm-qpjSITHAdexGc1zxM1gCIBx3TRg6yQt0lgafj52Ez4EbNwC-BREHgp6Do9w8am5i&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgRBn74XcIYWREnve1xcimboP7wYKYIBfEWFkZfUd-GYICICZk-ctk6yWr8swfA34qc3BWMMIvCxDtOpJNwQsy5G8l"
[dashsegments] Total fragments: 32
[download] Destination: 【零 ～月蝕の仮面～】ホラーゲームをそんなに怖がらないあの風見は、バケモンをスナップさせていただきます。【風見くく ⧸ あにまーれ】 [sJrDp5UeRIM].webm
[debug] File locking is not supported. Proceeding without locking
[download]   0.1% of ~ 320.00MiB at   39.87KiB/s ETA 02:16:49 (frag 0/32)
ERROR: Interrupted by user
```

Interestingly, I have not been able to get the throttling to occur with any videos that are a day old or older on the channels I am scanning, but many of the videos uploaded today are a problem.

**Comment by fren-archivist** [comment]
@Vangelis66  @0xcc07c9 The throttling may be inconsistent; I have seen the same format get both fast and slow download speeds just by retrying. I just tried 7t0SqerlBA0 several times. With format 315 I was able to get both throttled and unthrottled download speeds by just re-extracting and retrying. With 299 I only saw slow speeds though I only tried 5 times.

**Comment by Vangelis66** [comment]
Well, if the devs can't reproduce the issue themselves in a permanent/reliable fashion, debugging this will be really difficult...

**Comment by bashonly** [comment]
> I am also seeing this issue on a number of videos which are not age-restricted for both audio and video formats. I am using the patched version 2023.03.01.19419. It is a problem on most new videos for me, though somewhat inconsistent: sometimes only one format is throttled, sometimes both are, sometimes restarting the download solves the problem but sometimes it does not. 
> `[debug] Command-line config: ['sJrDp5UeRIM', '-f', '243', '-vU']`
> Interestingly, I have not been able to get the throttling to occur with any videos that are a day old or older on the channels I am scanning, but many of the videos uploaded today are a problem.

The pattern that's emerging is that throttling vs no throttling depends on the player client. Format 243 of your example video is only available via the `web` client. yt-dlp uses the `tv_embedded` client to circumvent the age-gate. Both of these clients seem to be throttled, with `tv_embedded` seemingly always throttled.

A workaround for now is to use `--extractor-args "youtube:player_client=android"`, since `android` client downloads seem to be working fine for now. Note that some formats may be missing when using this workaround, and yt-dlp's built-in age-gate bypass will not work.

**Comment by coletdjnz** [comment]
It appears to be all JS-based clients that are being throttled. This looks very similar to the old n-param throttling issue we had.

`android`, `ios` clients are fine

**Comment by rockxwre** [comment]
Using the latest version (2023.03.03.19419) it does not support protocol _https_ anymore for this video: https://www.youtube.com/watch?v=2fAalDg6sKE

I used the command 

```yt-dlp -vU -f "ba*[ext=m4a][protocol=https]" https://www.youtube.com/watch?v=2fAalDg6sKE``` but is results in the error
```
Latest version: 2023.03.03.19419, Current version: 2023.03.03.19419
yt-dlp is up to date (2023.03.03.19419)
[youtube] Extracting URL: https://www.youtube.com/watch?v=2fAalDg6sKE
[youtube] 2fAalDg6sKE: Downloading webpage
[youtube] 2fAalDg6sKE: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
ERROR: [youtube] 2fAalDg6sKE: Requested format is not available. Use --list-formats for a list of available formats
```

Omitting the protocol in the command does solve the slow data transfer issue though.
Using the latest stable version (2023.02.17) _does_ support the _https_ protocol, but the download is very slow.

**Comment by pukkandan** [comment]
> Using the latest version (2023.03.03.19419) it does not support protocol _https_ anymore for this video: [youtube.com/watch?v=2fAalDg6sKE](https://www.youtube.com/watch?v=2fAalDg6sKE)

That is expected. If we don't convert the formats to DASH, youtube will throttle them

**Comment by Vangelis66** [comment]
Immense thanks for the fix! 🥇 ... And I don't want to "blow my own horn" 😜 , but, consistent with:  

> edit: appears to be **the n-param decoding wrong**

I did hint in a [previous comment](https://github.com/yt-dlp/yt-dlp/issues/6400#issuecomment-1452805625) (that I've since hidden) that 

> Perhaps new player `7862ca1f` is somehow relevant?

Regards and thanks 😄 !

**Comment by pukkandan** [comment]
Tip for debugging in future:

When YT throttles, the first thing to check for is `--extractor-args youtube:player-client=android`. If it works, the problem is most likely that nsig broke. This can be verified by extracting relevant code manually[^2] from the player JS, running it in browser console and comparing with the value yt-dlp gives[^1].

Once we determine that the issue is with n-sig, the specific problem can be identified by stepping through the JS code with node/jsinterp in parallel and checking where they differed. Then all we are left with is actually fixing it[^3].

Debugging https://github.com/yt-dlp/yt-dlp/issues/6369 was more involved since it affected android formats as well and was unrelated to n-sig. None of the developers being unable to reproduce for the first day definitely didn't help either.

[^1]: player id, encrypted sig and result are all visible in verbose log
[^2]: See https://github.com/yt-dlp/yt-dlp/blob/9acf1ee25f7ad3920ede574a9de95b8c18626af4/yt_dlp/extractor/youtube.py#L3074-L3078
[^3]: Which turned out to be easy this time, but not in #1437, #4635

**Comment by ultrasound1372** [comment]
Slightly OT but a good comment for people wanting to test. Instead of telling pip to install directly from the git master, clone the git master, change into it and do
```
pip install -e .
```
Then when new updates come in, simply execute `git pull` within this directory, and wham latest patches are there! I personally use the awesome [pipx](https://pypa.github.io/pipx/) for this, where the `-e` option still works.

**Comment by skdursh** [comment]
Haven't tested every option but chiming in to say that with the most recent release (https://github.com/yt-dlp/yt-dlp/commit/354d5fca7a533f60a6eee14e854b9b4c132a7314) I'm also getting throttling with -f 140.

**Comment by bashonly** [comment]
@skdursh show verbose log `-vU`

**Comment by skdursh** [comment]
@bashonly how does -vU work? Should it print a logfile to the destination folder or is the relevant info in the console? I see a debug comment in the console with a gibberish url, but nothing that looks relevant to the issue.

**Comment by bashonly** [comment]
@skdursh the debug information is used for debugging. If you want to get to the bottom of this issue, please post all of it

**Comment by CoffeSiberian** [comment]
hey, in my case I use the URL generated by the request directly to use my own download methods to which I assigned a header ('Range': 'bytes = 0-') to have more download speed but this way no longer It works since the last updates.
Any suggestions?

**Comment by chrcoluk** [comment]
I had another 6 videos in my list to download, so I downloaded them using the fixed binary from the other issue.

5 of them were slow on the video, but fast on the audio download. (bits/second measured not time)
1 of them was fast on both.

However they were all old videos likely not in fast CDN's.

**Comment by pukkandan** [comment]
If more people keep reporting "issues" without showing logs, I'll just lock this thread too

**Comment by gatecrasher777** [comment]
Working on another YouTube project, today I managed to resolve all my recently experienced speed issues by adding ```&range=start-end``` to all (deciphered & n- transformed) stream urls.

Previously, you didn't need to have a range parameter in the urls, now you do otherwise the streams can be severely throttled. Having range headers in your download requests is not a substitute.

If you are using stream urls in html video or audio tags, as I do, just add ```&range=0-clen``` to the urls
with ```clen``` being the content length parameter of the same stream urls.

Then everything is fast again.

**Comment by NewUserHa** [comment]
Tested with the url from https://github.com/yt-dlp/yt-dlp/issues/6400#issuecomment-1453335114:
`yt-dlp version 2023.02.17 [a0a7c01]` w/ or w/o `'youtube:player-client=android'` both has the same about 157KiB/s.
`yt-dlp version 2023.03.03 [934496428]` w/ or w/o `'youtube:player-client=android'` both has the same up to about 3MiB/s (from 10kb/s speed up to).
all tests with `-vvvvvv`, but none of them print `n-sign`, so didn't include any in this result as well.

BTW, what is the difference between `https` and `dash` protocols for the same video/audio ID?
Is that `dash` uses hls m3u8 of multi-parted webm/mp4 files, but `https` uses from a single webm/mp4 file? Because usually, encoded to a single video file has fewer keyframes than encoded to multi-parted. (except cutting on stream and remuxing).

**Comment by pukkandan** [comment]
> BTW, what is the difference between `https` and `dash` protocols for the same video/audio ID?

See https://github.com/yt-dlp/yt-dlp/issues/6400#issuecomment-1454903800. Changing only Range headers to a single URL is the normal method of downloading files and is what ytdl/p calls "https" protocol. Youtube requires that we change the `query` parameter instead, effectively having to request multiple URLs per video. We call the result of each of the URL a "fragment" and the proces the "dash" protocol. This is similar to, but not the same as m3u8/hls. If you want to know detailed differences, I suggest you google it - these are both standard protocols used in video streaming.

**Comment by NewUserHa** [comment]
Right, it's a name that can easily get ppl confused with 'Dynamic Adaptive Streaming over HTTP (DASH)', but I already found those in the codes and updated the post, see the "Edited:".

BTW, if as the above one post said that `&range=0-clen` could work, then it can be better than `http_frag` because the single file only needs to speed up once, so higher downloading speed.

**Comment by pukkandan** [comment]
> it's a name that can easily get ppl confused with 'Dynamic Adaptive Streaming over HTTP (DASH)'

It is the same. Just that we are constructing the playlist instead of youtube providing it directly

**Comment by pukkandan** [comment]
> as the above one post said that `&range=0-clen` could work

We have tested that to not work for longer videos

**Comment by gatecrasher777** [comment]
> > as the above one post said that `&range=0-clen` could work
> 
> We have tested that to not work for longer videos

Appending `&range=0-clen` to the stream url works in an html video source tag, regardless of video length. Without it, buffering on playback is horrible in all browsers. Only very low quality streams will play without buffering. 

In a downloader, for a longer video, you would normally download it in chunks, so you would append `&range=start-end` to the url for each chunk. You would match these with the Range request header.

The point is that the YouTube player itself (base.js or embed.js) appends a &range parameter to the stream url of each video playback request, and the googlevideo.com servers (since recently) will delay and throttle requests if it is not included.

It might "not work" in every instance, but it is not going to break anything if it doesn't.

**Comment by dozyx** [comment]
> > as the above one post said that `&range=0-clen` could work
> 
> We have tested that to not work for longer videos

It seems work for longer videos now.

**Comment by ForHelvete** [comment]
Is there any quality loss from downloading the DASH-versions? It's always the same file size.

**Comment by bashonly** [comment]
@ForHelvete No, you end up with the same file, it's just been downloaded differently.

**Comment by NewUserHa** [comment]
yt-dlp could use a whole downloading of the URL rather than a fragmented downloading of it if as above said that now `&range=0-clen` can work.
And that will get better speed.

Should change it?

**Comment by pukkandan** [comment]
We are aware. But we want to wait and see if youtube adds throttling again. If they don't revert the change till next release, I will change it. Until then, `--extractor-arg youtube:include_duplicate_formats` or https://github.com/bashonly/yt-dlp-YTNoDashy can be used with the nightly builds to get the non-dashy formats

**Comment by fireattack** [comment]
If we're going to revert it, could it be left as an option? I use `-N 10` all the time, but IIRC it only works on fragments, so with the traditional HTTP, downloading from YouTube is relatively slow here. 

But now with this http-to-dashy conversion/workaround, it looks to me `yt-dlp` can download even HTTP sources with multi-thread downloading, which speeds it up significantly .

**Comment by pukkandan** [comment]
The plan is to only revert the default. `--extractor-arg youtube:include_duplicate_formats` will continue to list both sets of formats. There was some discussion for adding another switch for just enabling dashy formats, idk if that's useful. 

> I use `-N 10` all the time, but IIRC it only works with fragmented download

I am aware of this use-case. It's funny how we never thought to implement this till youtube forced our hand.

> (Or just make `-N` work with HTTP download by using `range` like various downloaders.)

Our http downloader code is a bit complex and I don't have the expertise to properly implement this. There is also not too much incentive to implement it since --external-downloader aria2c etc can be used for it.

**Comment by bughit** [comment]
This has regressed, for example https://github.com/yt-dlp/yt-dlp/issues/6400#issuecomment-1452718070 audio (140) download is throttled to 70-80 KiB/s

**Comment by pukkandan** [comment]
See #6131
